tests/ui_tools/test_boxes.py

```diff
@@ -530,6 +531,22 @@ def test_reset_search_text(self, panel_search_box):
         assert panel_search_box.caption == ""
         assert panel_search_box.edit_text == panel_search_box.search_text
 
+    @pytest.mark.parametrize("search_text, entered_string, expected_result", [
+        param('', 'backspace', False, id="no_text-disallow_urwid_backspace"),
+        param('', 'x', True, id="no_text-allow_entry_of_x"),
+        param('', ' ', False, id="no_text-disallow_entry_of_space"),
+        param('x', ' ', True, id="text-allow_entry_of_space"),
+        # backspace is not *entered*, but treated specially when passed in
```

Comment:
Could you clarify what do you mean by 'backspace is not *entered*'? The backspace does seem to get entered into the search box when some text is already present and get validated by `return super().valid_char(ch)`.

Reply:
If you experiment with `valid_char` just returning `True`, you'll notice that special urwid keys like `enter` and `backspace` are entered as text too, rather than treated as unicode characters. This is maybe mentioned more in the code than the test?

Reply:
Ah! Thanks for the clarification.